buildOutput.js

```diff
@@ -1,57 +1,166 @@
 const fs = require('fs');
 const path = require('path');
+const { ethers } = require('ethers');
 
 // Directory containing the JSON files
-const directoryPath = './chains'; // Update this path as needed
+const directoryPath = './chains';
 
-// Function to read and process files
-function processFiles(directoryPath) {
-	fs.readdir(directoryPath, (err, files) => {
-		if (err) {
-			console.error("Could not list the directory.", err);
-			process.exit(1);
+// Helper function to verify RPC endpoint
+async function verifyRPC(rpcUrl, expectedChainId) {
+	try {
+		const provider = rpcUrl.startsWith('wss://')
+			? new ethers.WebSocketProvider(rpcUrl)
+			: new ethers.JsonRpcProvider(rpcUrl);
+
+		const networkChainId = (await provider.getNetwork()).chainId;
+
+		if (rpcUrl.startsWith('wss://')) {
+			await provider.destroy();
+		}
+
+		const isValid = BigInt(networkChainId) === BigInt(expectedChainId);
+		if (!isValid) {
+			console.error(`RPC ${rpcUrl} returned wrong chain ID: ${networkChainId} (expected ${expectedChainId})`);
+		}
+		return isValid;
+	} catch (error) {
+		console.error(`Failed to verify RPC ${rpcUrl}:`, error.message);
+		return false;
+	}
+}
+
+// Helper function to verify URL accessibility
+async function verifyURL(url) {
+	try {
+		const response = await fetch(url);
+		return response.status === 200;
+	} catch (error) {
+		console.warn(`Failed to verify URL ${url}:`, error.message);
+		return false;
+	}
+}
+
+// Helper function to verify explorer
+async function verifyExplorer(explorer) {
+	try {
+		// First verify the base URL
+		const baseResponse = await fetch(explorer.url);
+		if (baseResponse.status !== 200) {
+			console.warn(`Explorer base URL ${explorer.url} is not accessible`);
+			return false;
 		}
+
+		// If it's an EIP3091 explorer, also verify block endpoint
+		if (explorer.standard === 'EIP3091') {
+			const blockUrl = `${explorer.url}/block/1`;
+			const blockResponse = await fetch(blockUrl);
+			if (blockResponse.status !== 200) {
+				console.warn(`Explorer block endpoint ${blockUrl} is not accessible`);
+				return false;
+			}
+		}
+
+		return true;
+	} catch (error) {
+		console.warn(`Failed to verify explorer ${explorer.url}:`, error.message);
+		return false;
+	}
+}
+
+// Helper function to verify dispatcher address
+async function verifyDispatcher(dispatcherAddr, rpcUrl) {
```

Comment:
I dont't think we need to verify dispatcher addresses. We may want to update the registry with CROSS_L2_PROVER_ADDRESS though

Reply:
That's an old verification script that has been used for a while. Since cross l2 prover is a new contract I planned to add it in a new PR with deprecation of dispatcher addresses.